src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -66,41 +68,105 @@ public static string Encode(string arg)
         /// <param name="offset">The number of elements of inArray to convert.</param>
-        /// <returns>The string representation in base 64 url encodingof length elements of inArray, starting at position offset.</returns>
+        /// <returns>The string representation in base 64 url encoding of length elements of inArray, starting at position offset.</returns>
         /// <exception cref="ArgumentNullException">'inArray' is null.</exception>
         /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
         public static string Encode(byte[] inArray, int offset, int length)
         {
             if (inArray == null)
                 throw LogHelper.LogArgumentNullException("inArray");
 
-            string s = Convert.ToBase64String(inArray, offset, length);
-            s = s.Split(base64PadCharacter)[0]; // Remove any trailing padding
-            s = s.Replace(base64Character62, base64UrlCharacter62);  // 62nd char of encoding
-            s = s.Replace(base64Character63, _base64UrlCharacter63);  // 63rd char of encoding
-            return s;
+            return EncodeString(Convert.ToBase64String(inArray, offset, length));
         }
 
         /// <summary>
         /// Converts a subset of an array of 8-bit unsigned integers to its equivalent string representation that is encoded with base-64-url digits. Parameters specify
         /// the subset as an offset in the input array, and the number of elements in the array to convert.
         /// </summary>
         /// <param name="inArray">An array of 8-bit unsigned integers.</param>
-        /// <returns>The string representation in base 64 url encodingof length elements of inArray, starting at position offset.</returns>
+        /// <returns>The string representation in base 64 url encoding of length elements of inArray, starting at position offset.</returns>
         /// <exception cref="ArgumentNullException">'inArray' is null.</exception>
         /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
         public static string Encode(byte[] inArray)
         {
             if (inArray == null)
                 throw LogHelper.LogArgumentNullException("inArray");
 
-            string s = Convert.ToBase64String(inArray, 0, inArray.Length);
-            s = s.Split(base64PadCharacter)[0]; // Remove any trailing padding
-            s = s.Replace(base64Character62, base64UrlCharacter62);  // 62nd char of encoding
-            s = s.Replace(base64Character63, _base64UrlCharacter63);  // 63rd char of encoding
+            return EncodeString(Convert.ToBase64String(inArray, 0, inArray.Length));
+        }
+
+        internal static string EncodeString(string str)
+        {
+#if NET45
+            str = str.Split(base64PadCharacter)[0]; // Remove any trailing padding
```

Comment:
do we need to check that the length is greater than 0 in-addition to not null?

Reply:
that's interesting, i think it's ok as an empty array, will return empty string.